packages/block-editor/src/components/iframe/index.js

```diff
@@ -304,10 +296,9 @@ function Iframe( {
 				} }
 				ref={ useMergeRefs( [ ref, setRef ] ) }
 				tabIndex={ tabIndex }
-				// Correct doctype is required to enable rendering in standards
-				// mode. Also preload the styles to avoid a flash of unstyled
-				// content.
-				src={ src }
+				// Using srcDoc instead of blob URL src to preserve referrer
+				// context in Safari, allowing YouTube embeds to work correctly.
+				srcDoc={ srcDoc }
```

Comment:
As noted in the comment, using `srcDoc` rather than a Blob URL preserves the referrer in Safari.